packages/terra-consumer-nav/src/components/nav-items/NavItem.jsx

```diff
@@ -54,20 +54,24 @@ const NavItem = ({
   ...customProps
 }) => {
   const activeClass = cx('active');
+  const badgeString = `${badgeValue}`;
 
-  const itemLabel = (<div>
-    <SafeHtml text={text} />
-    { badgeValue > 0 &&
-      <div className={cx('badge')}>{badgeValue}</div>
-    }
-  </div>);
+  const itemLabel = (
+    <div>
+      <SafeHtml text={text} />
+      { (!!badgeValue && badgeString.trim()) &&
```

Comment:
won't this fail on `0`?

Reply:
yup, that's the point though, unless we want to display badge that has 0. That, to me, seems counter-productive.

Reply:
yeah we want to display whatever value is passed unless its null or empty string.

Reply:
okay will update this